tesseract_environment/src/core/environment.cpp

```diff
@@ -1158,6 +1216,34 @@ bool Environment::applyRemoveJointCommand(const RemoveJointCommand::ConstPtr& cm
   return true;
 }
 
+bool Environment::applyReplaceJointCommand(const ReplaceJointCommand::ConstPtr& cmd)
+{
+  tesseract_scene_graph::Joint::ConstPtr current_joint = scene_graph_->getJoint(cmd->getJoint()->getName());
+  if (current_joint == nullptr)
+  {
+    CONSOLE_BRIDGE_logWarn("Tried to replace Joint (%s) that does not exist", cmd->getJoint()->getName().c_str());
+    return false;
+  }
+
+  if (cmd->getJoint()->child_link_name != current_joint->child_link_name)
+  {
+    CONSOLE_BRIDGE_logWarn("Tried to replace Joint (%s) where the child links are not the same",
+                           cmd->getJoint()->getName().c_str());
+    return false;
+  }
+
+  if (!scene_graph_->removeJoint(cmd->getJoint()->getName()))
+    return false;
+
+  if (!scene_graph_->addJoint(*cmd->getJoint()))
+    return false;
```

Comment:
If this fails, should we re-add the original joint (if that is possible)? At the least we should print an error letting the user know that the original joint was removed successfully since that won't be stored in the commands

Reply:
Good catch. Fixed.